src/app/model/Pawn.ts

```diff
@@ -0,0 +1,73 @@
+import { Board } from './Board';
+import { Piece } from './Piece';
+import { PieceMovement } from './PieceMovement';
+import { Column, columns, Row, Side, Square } from './Types';
+
+const NEXT_ROW_DIFFERENCE = {
+  WHITE: 1,
+  BLACK: -1,
+};
+
+export class Pawn extends Piece implements PieceMovement {
+  private nextRowDifference = 1;
+
+  constructor(id: string, side: Side) {
+    super('uselessId', side);
+    if (this.side !== 'WHITE') this.nextRowDifference = -1;
+    this.nextRowDifference = this.side === 'WHITE' ? NEXT_ROW_DIFFERENCE.WHITE : NEXT_ROW_DIFFERENCE.BLACK;
+  }
+
+  possibleMoves(position: Square, board: Board): Square[] {
+    let movesToGo: Square[] = [] as Square[];
+    movesToGo = movesToGo.concat(this.goAhead(position, board), this.goDoubleAhead(position, board), this.goDiagonalAhead(position, board));
+    return movesToGo;
+  }
```

Comment:
Chyba musiałeś mieć Piotrek zaćmienie umysłu :) Bo można to tak najprościej chyba:
```suggestion
  possibleMoves(position: Square, board: Board): Square[] {
    return this.goAhead(position, board).concat(this.goDoubleAhead(position, board), this.goDiagonalAhead(position, board))
  }
```

A używając spread operator to tak:

```suggestion
  possibleMoves(position: Square, board: Board): Square[] {
    return [...this.goAhead(position, board), ...this.goDoubleAhead(position, board), ...this.goDiagonalAhead(position, board)]
  }
```

Co kto woli :)